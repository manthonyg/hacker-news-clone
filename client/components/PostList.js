import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import moment from 'moment';
import PostSkeleton from './PostSkeleton';

const List = styled.ol`
  list-style: none;
  counter-reset: item;
  margin-bottom: 20px;
  max-width: 600px;
`;

const ListTitle = styled.span`
  font-weight: 700;
  font-size: 1em;
  color: #bb86fc;
`;
const ListItem = styled.li`
  margin: 2em;
  padding-left: 25px;
  position: relative;
  margin-bottom: 10px;

  &: before {
    content: '${props => {
      if (props.link) return 'ðŸ”—';
      if (props.comments > 50) return 'ðŸ”¥';
      if (props.link && props.comments > 20) return 'ðŸ”—ðŸ”¥';
      if (props.comments === 0) return 'ðŸ§Š';
      return '';
    }}';
    counter-increment: item;
    position: absolute;
    left: 0;
    top: 0;
    width: 20px;
    height: 20px;
    line-height: 20px;
    text-align: center;
    font-size: 10px;
    color: #fff;
    border-radius: 50%;
    text-indent: 0;
  }
  &: hover {
    background-color: #ffc;
  }
`;

function PostList({ posts, isLoading }) {
  return (
    <>
      {isLoading ? (
        <PostSkeleton numberOfSkeletons={10} />
      ) : (
        <List>
          {posts &&
            !!posts.length &&
            posts.map(post => {
              return (
                <>
                  <ListItem
                    comments={post && post.kids && post.kids.length}
                    link={post.url}
                  >
                    <ListTitle>
                      <a href={post.url}>{post.title}</a>
                    </ListTitle>
                    <br />
                    <span>
                      <Link to={`/user?id=${post.by}`}>by {post.by}</Link>
                      at {moment.unix(post.time).format('YYYY-MM-DD')}
                    </span>

                    <Link to={`/post?id=${post.id}`}>
                      {post.descendants} comments
                    </Link>
                  </ListItem>
                </>
              );
            })}
        </List>
      )}
    </>
  );
}

export default PostList;
