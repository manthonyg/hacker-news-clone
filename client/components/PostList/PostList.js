import React from "react";
import styled, { keyframes } from "styled-components";
import { Link } from "react-router-dom";
import moment from "moment";
import { truncateString } from "../../utils/truncate";
import Post from "../Post/Post";

function PostList({ posts }) {
  console.log(posts);
  return (
    <>{posts && posts.length && posts.map((post) => <Post post={post} />)}</>
  );
}

export default PostList;
