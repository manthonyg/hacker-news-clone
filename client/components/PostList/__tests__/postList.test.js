import React from "react";
import renderer from "react-test-renderer";
import PostList from "../../PostList/PostList";
import RenderWithRouter from "../../../test_utils/renderWithRouter";

it("renders correctly with all props", () => {
  const requiredProps = {
    posts: [
      {
        id: 5,
        url: "https://www.sitkneelbend.com",
        score: 999,
        kids: [],
        by: "moopie",
        title: "Jest: a tale of misery",
        time: 2392031230,
      },
    ],
  };
  const tree = renderer
    .create(
      <>
        <RenderWithRouter>
          <PostList {...requiredProps} />
        </RenderWithRouter>
      </>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
