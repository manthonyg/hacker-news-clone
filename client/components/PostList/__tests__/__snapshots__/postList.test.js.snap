// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<div
  className="sc-AxjAm bArVMS"
  data-testid="POST"
>
  <ol
    className="sc-AxirZ eJDvGb"
  >
    <li
      className="sc-AxgMl bcugQl"
    >
      <span
        className="sc-AxhCb hKagXQ"
      >
        <a
          href="https://www.sitkneelbend.com"
        >
          Jest: a tale of misery
        </a>
      </span>
      <p
        className="sc-AxhUy kKGMVQ"
      >
        (
        https://www.sitkneelbend.com
        )
      </p>
      <br />
      <span
        className="sc-AxheI kchDLz"
      >
        <a
          className="sc-AxiKw dVyqFv"
          href="/user?id=moopie"
          onClick={[Function]}
        >
          by 
          moopie
        </a>
         | 
        2045-10-19
      </span>
    </li>
  </ol>
</div>
`;
